src/Sarif.FunctionalTests/SpecExamples/Recommended.sarif

```diff
@@ -0,0 +1,55 @@
+{
+  "version": "1.0.0",
+  "runs": [
+    {
+      "tool": {
+        "name": "CodeScanner",
+        "semanticVersion": "2.1.0"
+      },
+      "files": {
+        "file:///user/builder/work/src/collections/list.cpp": {
+          "mimeType": "text/x-c"
+        }
+      },
+      "logicalLocations": {
+        "collections::list::add": {
+          "name": "add",
+          "kind": "function",
+          "parentKey": "collections::list"
+        },
+        "collections::list": {
+          "name": "list",
+          "kind": "type",
+          "parentKey": "collections"
+        },
+        "collections": {
+          "name": "collections",
+          "kind": "namespace"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "C2001",
+          "message": "Variable \"n\" was used without being initialized.",
+          "locations": [
+            {
+              "analysisTarget": {
+                "uri": "file:///user/builder/work/src/collections/list.cpp",
+                "region": {
+                  "startLine": 15
```

Comment:
Oh, and to answer your question: `startLine` is required for a text file (assuming the result has a region at all, which it need not). But my "minimal" example, which by design is the absolutely minimal valid SARIF file, doesn't have any results in it.

I wouldn't spend a lot of effort deciding exactly what goes in each example. The problem is nesting. A minimal file doesn't have to have any `result`s. If it does have a `result`, the `result` doesn't have to have any `location`s. If it does have a `location`, then the `physicalLocation` (that is, either the `analysisTarget` or the `resultFile`) doesn't have to have a `region`. If it does have a `region`, _and_ it's in a text file -- ahh, _then_ `startLine` is required. It isn't practical to show examples with each level of nesting.

While I'm on the topic of the difficulty of crafting these examples: I can't give a really "comprehensive" example, either, in the sense of #35 -- that is, an example that demonstrates every property -- because some properties are mutually exclusive. For example, a `result` can't have both a `message` and a `formattedRuleMessage` (I'll demonstrate `message` in the "recommended" example and `formattedRuleMessage` in the "comprehensive" example).

Bottom line, I consider this a "best effort" exercise, with the goal of showing implementers realistic examples. But it can't be exhaustive.